File: PakClassified.WebApp/PakClassified.WebApp.DTOs/PakClassified.DTOs/AdvertisementModel.cs
using a._PakClassified.WebApp.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PakClassified.WebApp.DTOs.PakClassified.DTOs
{
    public class AdvertisementModel: IEntityNamedModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public decimal Price { get; set; }
        public int? Likes { get; set; }
        public DateTime StartsOn { get; set; }
        public DateTime EndsOn { get; set; }
        public string CreatedBy { get; set; }
        public string? LastModifiedBy { get; set; }
        public int CityAreaId { get; set; }
        public int PostedById { get; set; }
        public int StatusId { get; set; }
        public int TypeId { get; set; }
        public int SubCategoryId { get; set; }
        public ICollection<AdvertisementTagModel> Tags { get; set; }
        public ICollection<AdvertisementImageModel> Images { get; set; }
    }
}

File: PakClassified.Client/src/app/core/models/pakClassified/advertisement-model.ts
import { UserGet } from "../user/user-model";

export interface AdvertisementGetPost {
    id: number;
    name: string;
    title: string;
    description?: string;
    price: number;
    likes: number;
    startsOn: Date;
    endsOn: Date;
    createdBy: string;
    lastModifiedBy?: string;
    cityAreaId: number;
    postedById: number;
    statusId: number;
    typeId: number;
    subCategoryId: number;
    tagsId: number[];
    imagesId: number[];
}
export interface Advertisement extends AdvertisementGetPost {
    cityArea?: string;
    postedBy?: UserGet;
    status?: string;
    type?: string;
    subCategory?: string;
    tagNames?: string[];
}